compair/api/group_user.py

```diff
@@ -0,0 +1,268 @@
+
+        return {'success': True}
+
+api.add_resource(GroupUserDeleteAPI, '/users/<user_uuid>')
+
+class GroupUserListDeleteAPI(Resource):
+    @login_required
+    def post(self, course_uuid):
+        # delete multiple (DELETE request cannot use params in angular)
+        course = Course.get_active_by_uuid_or_404(course_uuid)
+
+        require(EDIT, UserCourse(course_id=course.id),
+            title="Group Not Saved",
+            message="Sorry, your role in this course does not allow you to save groups.")
+
+        params = user_list_parser.parse_args()
+
+        if len(params.get('ids')) == 0:
+            abort(400, title="Group Not Saved", message="Please select at least one user below and then try to apply the group again.")
+
+        user_courses = UserCourse.query \
+            .join(User, UserCourse.user_id == User.id) \
+            .filter(and_(
+                UserCourse.course_id == course.id,
+                User.uuid.in_(params.get('ids')),
+                UserCourse.course_role != CourseRole.dropped
+            )) \
+            .all()
+
+        if len(params.get('ids')) != len(user_courses):
+            abort(400, title="Group Not Saved", message="One or more users selected are not enrolled in the course yet.")
+
+        for user_course in user_courses:
+            if course.groups_locked and user_course.group_id != None:
+                abort(400, title="Group Not Saved",
+                    message="The course groups are locked. You may not remove users from the group they are already assigned to.")
+
+        for user_course in user_courses:
+            user_course.group_id = None
+
+        db.session.commit()
+
+        on_group_user_list_delete.send(
+            current_app._get_current_object(),
+            event_name=on_group_user_list_delete.name,
+            user=current_user,
+            course_id=course.id,
+            data={'user_ids': [user_course.user_id for user_course in user_courses] }
+        )
+
+        return { 'success': True }
+
+api.add_resource(GroupUserListDeleteAPI, '/users')
+
+
+class GroupCurrentUserAPI(Resource):
+    @login_required
+    def get(self, course_uuid):
+        course = Course.get_active_by_uuid_or_404(course_uuid)
+
+        require(READ, UserCourse(user_id=current_user.id),
+            title="Group Unavailable",
+            message="Groups can be seen those enrolled in the course. Please double-check your enrollment in this course.")
+
+        user_course = UserCourse.query \
+            .filter(and_(
+                UserCourse.course_id == course.id,
+                UserCourse.user_id == current_user.id,
+                UserCourse.course_role != CourseRole.dropped
+            )) \
+            .one_or_none()
+
+        if not user_course:
+            abort(400, title="Group Unavailable",
+                message="You are not currently enrolled in the course. Please double-check your enrollment in this course.")
```

Comment:
Admin users can't open / view assignment because of this check

Reply:
Thanks! I was also missing tests for the current user's group endpoint in general